ngraph/frontend/frontend_manager/include/frontend_manager/frontend_manager.hpp

```diff
@@ -0,0 +1,518 @@
+
+            /// \brief Loads an input model from set of memory buffers
+            /// Memory structure is frontend-defined and is not specified in generic API
+            /// \param modelParts Array of model memory buffers
+            /// \return Loaded input model
+            virtual InputModel::Ptr
+                load_from_memory_fragments(const std::vector<const void*>& modelParts) const;
+
+            /// \brief Loads an input model by input stream representing main model file
+            /// \param stream Input stream of main model
+            /// \return Loaded input model
+            virtual InputModel::Ptr load_from_stream(std::istream& stream) const;
+
+            /// \brief Loads an input model by input streams representing all model files
+            /// \param streams Array of input streams for model
+            /// \return Loaded input model
+            virtual InputModel::Ptr
+                load_from_streams(const std::vector<std::istream*>& streams) const;
+
+            /// \brief Completely convert and normalize entire function, throws if it is not
+            /// possible
+            /// \param model Input model
+            /// \return fully converted nGraph function
+            virtual std::shared_ptr<ngraph::Function> convert(InputModel::Ptr model) const;
+
+            /// \brief Completely convert the remaining, not converted part of a function.
+            /// \param partiallyConverted partially converted nGraph function
+            /// \return fully converted nGraph function
+            virtual std::shared_ptr<ngraph::Function>
+                convert(std::shared_ptr<ngraph::Function> partiallyConverted) const;
+
+            /// \brief Convert only those parts of the model that can be converted leaving others
+            /// as-is. Converted parts are not normalized by additional transformations; normalize
+            /// function or another form of convert function should be called to finalize the
+            /// conversion process.
+            /// \param model Input model
+            /// \return partially converted nGraph function
+            virtual std::shared_ptr<ngraph::Function>
+                convert_partially(InputModel::Ptr model) const;
+
+            /// \brief Convert operations with one-to-one mapping with decoding nodes.
+            /// Each decoding node is an nGraph node representing a single FW operation node with
+            /// all attributes represented in FW-independent way.
+            /// \param model Input model
+            /// \return nGraph function after decoding
+            virtual std::shared_ptr<ngraph::Function> decode(InputModel::Ptr model) const;
+
+            /// \brief Runs normalization passes on function that was loaded with partial conversion
+            /// \param function partially converted nGraph function
+            virtual void normalize(std::shared_ptr<ngraph::Function> function) const;
```

Comment:
What if we want apply not only normalization transformations.

Reply:
@slyalin can you please comment on this? Per my understanding, we can't do much with partially converted function

Reply:
Any usual transformation can be applied on this function. FE::normalize method collects transformations that FE usually applies after function is converted. So when FE::convert is called, it internally calls all transformations from FE::normalize. Having FE::normalize as a separate function allows user to construct their own pipelines that consists of `FE::decode` or `convert_partially` and then `normalize` optionally mixing them with user-defined transformations.